nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/controllers/nf-ng-canvas-flow-status-controller.js

```diff
@@ -400,6 +406,554 @@
+                                // report the updated status
+                                previousRulesResponse = response;
+    
+                                // setup rule menu handling
+                                flowAnalysisCtrl.setRuleMenuHandling(response);
+
+                                // setup violation menu handling
+                                flowAnalysisCtrl.setViolationMenuHandling(response, groupId);
+                            }
+                        }).fail(nfErrorHandler.handleAjaxError);
+                    }
+                },
+
+                /**
+                 * Set event bindings for rule menus
+                 */
+                setRuleMenuHandling: function() {
+                    $('.rule-menu-btn').click(function(event) {
+                        // stop event from immediately bubbling up to document and triggering closeRuleWindow
+                        event.stopPropagation();
+                        var ruleInfo = $(this).data('ruleInfo');
+                        $('#violation-menu').hide();
+                        $('#rule-menu').show();
+                        $('#rule-menu').position({
+                            my: "left top",
+                            at: "left top",
+                            of: event
+                        });
+
+                        // rule menu bindings
+                        $('#rule-menu-more-info').on( "click", openRuleMoreInfoDialog);
+                        $('#rule-menu-edit-rule').on('click', openRuleDetailsDialog);
+                        $(document).on('click', closeRuleWindow);
+
+                        function closeRuleWindow(e) {
+                            if ($(e.target).parents("#rule-menu").length === 0) {
+                                $("#rule-menu").hide();
+                                unbindRuleMenuHandling();
+                            }
+                        }
+
+                        function openRuleDetailsDialog() {
+                            $('#rule-menu').hide();
+                            nfSettings.showSettings().done(function() {
+                                nfSettings.selectFlowAnalysisRule(ruleInfo.id);
+                            });
+                            unbindRuleMenuHandling()
+                        }
+
+                        function openRuleMoreInfoDialog() {
+                            $('#rule-menu').hide();
+                            $('#rule-type-pill').empty()
+                                                .removeClass()
+                                                .addClass(ruleInfo.enforcementPolicy.toLowerCase() + ' rule-type-pill')
+                                                .append(ruleInfo.enforcementPolicy);
+                            $('#rule-display-name').empty().append(ruleInfo.name);
+                            $('#rule-description').empty().append(ruleInfo.descriptors['component-type'].description);
```

Comment:
This refers to the description of the PropertyDescriptor named 'component-type'.
That is very specific to the DisallowComponentType rule.
If a rule doesn't have that descriptor this  fails with an error.

Important to not that some Rules don't have any PropertyDescriptors at all.

In general adding the PropertyDescriptors to this dialog page is not desirable in my opinion.

Reply:
Just to be clear, I think the dialog would be really valuable for a user if we did have descriptions for rules so they don't have to constantly look up documentation to find a short description for what a rule does, but if the property doesn't exist or isn't always present it would probably be better just to direct the user to the rule documentation.

Reply:
For now I'd go with directing the user to the rule documentation.